db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ScanCostFunction.java

```diff
@@ -147,6 +150,25 @@ private void addSelectivity(SelectivityHolder holder) {
         return holders;
     }
 
+    private void checkInListPredColumnStats(Predicate p) throws StandardException {
+        if (p.getSourceInList() != null) {
+            for (Object o : p.getSourceInList().leftOperandList) {
+                ColumnReference cr = (ColumnReference) o;
```

Comment:
Object from the leftOperandList of an inlist may not be a ColumnReference, for example: `a1+1 in (1,2,3)`. So it is better to be conservative to check the type before casting. The above code does not have problem now as currently, only inlist with column can come here.

Reply:
I see. Added a check before casting.